### PR TITLE
Add dynamodb table class support

### PR DESCRIPTION
**Pull request by marknet15**
##### SUMMARY
Add support for defining a `TableClass` on DynamoDB tables.

`TableClass` was introduced as part of `botocore` version `1.23.18`
https://github.com/boto/botocore/blob/develop/CHANGELOG.rst#12318

Fixes: https://github.com/ansible-collections/community.aws/issues/829

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
dynamodb_table

##### ADDITIONAL INFORMATION
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/dynamodb.html#DynamoDB.Client.create_table
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/dynamodb.html#DynamoDB.Client.update_table

### CODE REVIEW

File: plugins/modules/dynamodb_table.py [hunk 1]
```diff
@@ -121,6 +121,12 @@
     default: []
     type: list
     elements: dict
+  table_class:
+    description:
+      - The class of the table
```
Comment (alinabuzachis) [inline plugins/modules/dynamodb_table.py:126]:
```suggestion
      - The class of the table
      - Requires at least botocore version 1.23.18.
```

File: plugins/modules/dynamodb_table.py [hunk 2]
```diff
@@ -999,6 +1019,10 @@ def main():
     )
     client = module.client('dynamodb', retry_decorator=retry_decorator)
 
+    if module.params.get('table_class'):
+        if not module.botocore_at_least("1.23.18"):
+            module.fail_json(msg='botocore version >= 1.23.18 is required setting table_class')
+
```
Comment (alinabuzachis) [inline plugins/modules/dynamodb_table.py:1025]: I guess you can do something like this if you want.

```suggestion
    if module.params.get('table_class'):
        module.require_botocore_at_least('1.23.18', reason='to set table_class')
```

**Review by alinabuzachis — COMMENTED** [review]
LGTM, only some suggestions! May I ask you to update the RETURN statement please?

File: plugins/modules/dynamodb_table.py [hunk 3]
```diff
@@ -206,6 +213,11 @@
     returned: success
     type: str
     sample: ACTIVE
+table_class:
+    description: The current table class.
+    returned: when state present
+    type: str
+    sample: STANDARD_INFREQUENT_ACCESS
```
Comment (alinabuzachis) [inline plugins/modules/dynamodb_table.py:220]: This is the actual RETURN statement, I guess it should reflect this.
```
changed: [testhost] => {
    "billing_mode": "PROVISIONED",
    "changed": true,
    "hash_key_name": "id",
    "hash_key_type": "NUMBER",
    "indexes": [],
    "invocation": {},
    "range_key_name": null,
    "range_key_type": null,
    "read_capacity": 1,
    "region": "us-east-1",
    "resource_actions": [
        "dynamodb:ListTagsOfResource",
        "dynamodb:DescribeTable",
        "dynamodb:CreateTable"
    ],
    "table": {
        "arn": "arn:aws:dynamodb:us-east-1:721066863947:table/ansible-test-table",
        "attribute_definitions": [
            {
                "attribute_name": "id",
                "attribute_type": "N"
            }
        ],
        "billing_mode": "PROVISIONED",
        "creation_date_time": "2022-02-04T13:36:01.578000+00:00",
        "id": "533b45fe-0870-4b66-9b00-d2afcfe96f19",
        "item_count": 0,
        "key_schema": [
            {
                "attribute_name": "id",
                "key_type": "HASH"
            }
        ],
        "name": "ansible-test-14482047-alinas-mbp",
        "provisioned_throughput": {
            "number_of_decreases_today": 0,
            "read_capacity_units": 1,
            "write_capacity_units": 1
        },
        "size": 0,
        "status": "ACTIVE",
        "table_arn": "arn:aws:dynamodb:us-east-1:721066863947:table/ansible-test-table",
        "table_id": "533b45fe-0870-4b66-9b00-d2afcfe96f19",
        "table_name": "ansible-test-table",
        "table_size_bytes": 0,
        "table_status": "ACTIVE",
        "tags": {}
    },
    "table_class": null,
    "table_name": "ansible-test-table",
    "table_status": "ACTIVE",
    "tags": {},
    "write_capacity": 1
}

```

Comment (marknet15) [reply plugins/modules/dynamodb_table.py:220]: > This is the actual RETURN statement, I guess it should reflect this.
> 
> ```
> changed: [testhost] => {
>     "billing_mode": "PROVISIONED",
>     "changed": true,
>     "hash_key_name": "id",
>     "hash_key_type": "NUMBER",
>     "indexes": [],
>     "invocation": {},
>     "range_key_name": null,
>     "range_key_type": null,
>     "read_capacity": 1,
>     "region": "us-east-1",
>     "resource_actions": [
>         "dynamodb:ListTagsOfResource",
>         "dynamodb:DescribeTable",
>         "dynamodb:CreateTable"
>     ],
>     "table": {
>         "arn": "arn:aws:dynamodb:us-east-1:721066863947:table/ansible-test-table",
>         "attribute_definitions": [
>             {
>                 "attribute_name": "id",
>                 "attribute_type": "N"
>             }
>         ],
>         "billing_mode": "PROVISIONED",
>         "creation_date_time": "2022-02-04T13:36:01.578000+00:00",
>         "id": "533b45fe-0870-4b66-9b00-d2afcfe96f19",
>         "item_count": 0,
>         "key_schema": [
>             {
>                 "attribute_name": "id",
>                 "key_type": "HASH"
>             }
>         ],
>         "name": "ansible-test-14482047-alinas-mbp",
>         "provisioned_throughput": {
>             "number_of_decreases_today": 0,
>             "read_capacity_units": 1,
>             "write_capacity_units": 1
>         },
>         "size": 0,
>         "status": "ACTIVE",
>         "table_arn": "arn:aws:dynamodb:us-east-1:721066863947:table/ansible-test-table",
>         "table_id": "533b45fe-0870-4b66-9b00-d2afcfe96f19",
>         "table_name": "ansible-test-table",
>         "table_size_bytes": 0,
>         "table_status": "ACTIVE",
>         "tags": {}
>     },
>     "table_class": null,
>     "table_name": "ansible-test-table",
>     "table_status": "ACTIVE",
>     "tags": {},
>     "write_capacity": 1
> }
> ```

@alinabuzachis I am wondering if everything bar the actual `table` object and maybe `table_status` should be deprecated and future dropped as the `table` object contains all actual fields from the AWS describe call whereas the rest is basically a mirror of the inputted params.

Comment (marknet15) [reply plugins/modules/dynamodb_table.py:220]: Maybe if I doc just `table` for now ?

Comment (alinabuzachis) [reply plugins/modules/dynamodb_table.py:220]: You can do something like this in RETURN:

```
...
table:
    description: Add description
    type: complex
    sample: {
        "arn": "arn:aws:dynamodb:us-east-1:721066863947:table/ansible-test-table",
        "attribute_definitions": [
            {
                "attribute_name": "id",
                "attribute_type": "N"
            }
        ],
        "billing_mode": "PROVISIONED",
        "creation_date_time": "2022-02-04T13:36:01.578000+00:00",
        "id": "533b45fe-0870-4b66-9b00-d2afcfe96f19",
        "item_count": 0,
        "key_schema": [
            {
                "attribute_name": "id",
                "key_type": "HASH"
            }
        ],
        "name": "ansible-test-14482047-alinas-mbp",
        "provisioned_throughput": {
            "number_of_decreases_today": 0,
            "read_capacity_units": 1,
            "write_capacity_units": 1
        },
        "size": 0,
        "status": "ACTIVE",
        "table_arn": "arn:aws:dynamodb:us-east-1:721066863947:table/ansible-test-table",
        "table_id": "533b45fe-0870-4b66-9b00-d2afcfe96f19",
        "table_name": "ansible-test-table",
        "table_size_bytes": 0,
        "table_status": "ACTIVE",
        "tags": {}
    }
....
```

**Review by alinabuzachis — APPROVED** [review]
@marknet15 Thank you. LGTM!

**Review by jillr — APPROVED** [review]

**Pull request merged**